packages/std/src/ibc.rs

```diff
@@ -45,12 +52,35 @@ pub enum IbcMsg {
         data: Binary,
         /// when packet times out, measured on remote chain
         timeout: IbcTimeout,
+        /// When set, a [MsgPayPacketFee] with the provided fee is created and
+        /// emitted before the packet is sent.
+        /// `signer` gets filled automatically with the contract address.
+        /// No relayer restrictions will be in place.
+        ///
+        /// [MsgPayPacketFee]: https://github.com/cosmos/ibc-go/blob/v7.2.0/proto/ibc/applications/fee/v1/tx.proto#L76-L93
+        packet_fee: Option<IbcFee>,
     },
     /// This will close an existing channel that is owned by this contract.
     /// Port is auto-assigned to the contract's IBC port
     CloseChannel { channel_id: String },
 }
 
+/// An [ICS-29] IBC fee.
+///
+/// This is mapped to an [ibc.applications.fee.v1.Fee].
+///
+/// [ICS-29]: https://github.com/cosmos/ibc-go/blob/v7.2.0/docs/middleware/ics29-fee/overview.md
+/// [ibc.applications.fee.v1.Fee]: https://github.com/cosmos/ibc-go/blob/v7.2.0/proto/ibc/applications/fee/v1/fee.proto#L11-L31
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq, Eq, JsonSchema)]
+pub struct IbcFee {
+    /// The packet receive fee
+    pub recv: Vec<Coin>,
+    /// The packet acknowledgement fee
+    pub ack: Vec<Coin>,
+    /// The packet timeout fee
+    pub timeout: Vec<Coin>,
+}
```

Comment:
I left it out in order to keep this type simple. A developer can access to full API by adding a MsgPayPacketFee instead of using this field. Do we really need it here?